src/Model/Resolver/EntityUrl.php

```diff
@@ -139,7 +139,13 @@ public function resolve(
                 $collection = $this->productCollectionFactory->create()
                     ->addAttributeToFilter('status', ['eq' => Status::STATUS_ENABLED]);
                 $product = $collection->addIdFilter($id)->getFirstItem();
-                $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                $isInStock = false;
+
+                try {
+                    $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                } catch (NoSuchEntityException $e) {
+                    // Ignoring error is safe
```

Comment:
Instead of ignoring the error I would `return null;` but it is a preference. Merging.